### DISCUSSION TITLE
[express-fileupload] fileUpload.Options interface now requires 'headers' property

### DISCUSSION THREAD

**Discussion by Talb2005** [DefinitelyTyped/DefinitelyTyped#61875] [Issues with a @types package] [answered]
A [recent change](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/61835/files#diff-003a41e9e22231a63faef97ef660f61cc761a45f2b657b60e12f27a46cbb5f91L51) has introduced this issue by replacing `interface Options extends Partial<BusboyConfig>` with `interface Options extends BusboyConfig`.
Now `BusboyConfig.headers` is required when providing fileUpload.Options.. but I don't think it should be..

[a comment by an automated account is not shown]

**Reply by BendingBender** [reply]
@Talb2005 you should update your `@types/busboy` types or downgrade your `@types/express-fileupload` types, depending on which version of `express-fileupload` you use.

**Comment by Talb2005** [comment]
@BendingBender I’m not using busboy in my project.
I know I can downgrade my @types/express-fileupload, but that would just be a temporary fix..
The issue should be fixed in @types/express-fileupload as this `headers` option isn’t mentioned in express-fileupload.
I’m using express-fileupload 1.4.0 and @types/express-fileupload 1.4.0..

**Reply by BendingBender** [reply]
@Talb2005 `@types/express-fileupload` types depend on `@types/busboy` types. They are certainly installed in your project as one of the transient dependencies. Please check this and upgrade the types.

**Reply by Talb2005** [reply]
@BendingBender indeed installing latest @types/busboy solves the issue.
But I've also found a cleaner solution.. (no need to have @types/busboy in your project's `package.json`.)
Delete `package-lock.json` and run `npm install`. The latest @types/busboy would be installed and registered in `package-lock.json`.

**Reply by BendingBender** [reply]
That's exactly what I wanted you to try. Ok, I've made a PR to work around this problem, should other users run into it: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/61910

**Comment by peterblazejewicz** [comment, marked as answer]
this was fixed by #61910, wasn't it?

**Reply by BendingBender** [reply]
yes, should have been
